### PR TITLE
feat(component): add Table component

### PR DESCRIPTION
**Pull request by chanceaclark**
## What

Adding basic Table component.

A few notes:
- Documentation page is disabled for now; still a WIP.
- Needed the checkbox to render without a label. For now, just adding boolean as a property type so we could pass in false. Need to come up with a way to have a visually hidden label.
- Wrapped `Pagination` Dropdown in a Flex.Item for correct positioning.

## Screenshot
Screen Shot 2019-09-23 at 10 42 01 AM

### CODE REVIEW

File: packages/big-design/src/components/Table/Body/Body.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+import { TableSectionContext } from '../context';
+
+import { StyledTableBody } from './styled';
+
+export interface TableBodyProps extends React.TableHTMLAttributes<HTMLTableSectionElement> {}
+
+export const TableBody: React.FC<TableBodyProps> = ({ className, children, style, ...props }) => (
+  <TableSectionContext.Provider value={'tbody'}>
```
Comment (rtalvarez) [inline packages/big-design/src/components/Table/Body/Body.tsx]: 🍹  `value="tbody"`

File: packages/big-design/src/components/Table/Footer/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+import { TableSectionContext } from '../context';
+
+import { StyledTableFooter } from './styled';
+
+export type TableFooterProps = React.TableHTMLAttributes<HTMLTableSectionElement>;
+
+export const TableFooter: React.FC<TableFooterProps> = props => (
+  <TableSectionContext.Provider value={'tfoot'}>
```
Comment (rtalvarez) [inline packages/big-design/src/components/Table/Footer/Footer.tsx]: 🍹 same

File: packages/big-design/src/components/Table/Head/Head.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+import { TableSectionContext } from '../context';
+
+import { StyledTableHead } from './styled';
+
+export type TableHeadProps = React.TableHTMLAttributes<HTMLTableSectionElement>;
+
+export const TableHead: React.FC<TableHeadProps> = props => (
+  <TableSectionContext.Provider value={'thead'}>
```
Comment (rtalvarez) [inline packages/big-design/src/components/Table/Head/Head.tsx]: 🍹 same

File: packages/big-design/src/components/Table/Table.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+
+import { uniqueId } from '../../utils';
+
+import { TableContext } from './context';
+import { StyledTable, StyledTableFigure } from './styled';
+import { TableActions, TableActionsProps } from './Actions/Actions';
```
Comment (deini) [inline packages/big-design/src/components/Table/Table.tsx]: Can we create `index.ts` files for these imports?

File: packages/big-design/src/components/Table/Row/Row.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+export interface TableRowProps extends React.TableHTMLAttributes<HTMLTableRowElement> {
+  selected?: boolean;
+  verticalAlign?: 'top' | 'bottom' | 'middle';
+  onRowSelect?(selected: boolean): void;
+}
+
+export const TableRow: React.FC<TableRowProps> = memo(({ children, selected, onRowSelect, ...props }) => {
+  const tableContext = useContext(TableContext);
+  const tableSectionContext = useContext(TableSectionContext);
+  const [isSelected, setIsSelected] = useState(selected);
```
Comment (jorgemoya) [inline packages/big-design/src/components/Table/Row/Row.tsx]: I was under the impression `state` shouldn't be set from a `prop` received. Is this a pattern that has changed with hooks?

File: packages/big-design/src/components/Table/Cell/styled.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+  ${props =>
+    props.isCheckbox &&
+    css`
+      width: ${({ theme }) => theme.helpers.addValues(theme.spacing.xLarge, theme.spacing.small)};
+      white-space: nowrap;
+    `}
+
+  &:last-of-type {
+    width: 50%;
```
[the remark this answers is not in the document]
Comment (chanceaclark) [reply packages/big-design/src/components/Table/Cell/styled.tsx]: Good catch, that was stylings from the prop tables that need to be removed from this.

File: packages/big-design/src/components/Table/Table.tsx (hunk 4 above)
Comment (chanceaclark) [reply packages/big-design/src/components/Table/Table.tsx]: Is there a reason why we haven't been doing that for `Flex` and `Grid` static members then?

**Pull request merged**